frontend/src/components/GoalForm/GoalNudge.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Checkbox,
+} from '@trussworks/react-uswds';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { similiarGoalsByText } from '../../fetchers/goals';
+import { getGoalTemplates } from '../../fetchers/goalTemplates';
+import useAsyncDebounceEffect from '../../hooks/useAsyncDebounceEffect';
+import GoalNudgeText from './GoalNudgeText';
+import GoalNudgeInitiativePicker from './GoalNudgeInitiativePicker';
+
+const MINIMUM_GOAL_NAME_LENGTH = 15;
+
+export const filterOutGrantUsedGoalTemplates = (goalTemplates, selectedGrants) => goalTemplates
```

Comment:
Function to filter out curated goals from the options if one of the selected grants has a goal like that in flight